homeassistant/components/arve/sensor.py

```diff
@@ -0,0 +1,100 @@
+"""Sensor platform for Arve devices."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from asyncarve import ArveSensProData
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    CONCENTRATION_PARTS_PER_MILLION,
+    PERCENTAGE,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import ArveCoordinator
+from .entity import ArveDeviceEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class ArveDeviceEntityDescription(SensorEntityDescription):
+    """Describes Arve device entity."""
+
+    value_fn: Callable[[ArveSensProData], float | int]
+
+
+SENSORS: tuple[ArveDeviceEntityDescription, ...] = (
+    ArveDeviceEntityDescription(
+        key="CO2",
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        device_class=SensorDeviceClass.CO2,
+        value_fn=lambda arve_data: arve_data.co2,
+    ),
+    ArveDeviceEntityDescription(
+        key="AQI",
+        device_class=SensorDeviceClass.AQI,
+        value_fn=lambda arve_data: arve_data.aqi,
+    ),
+    ArveDeviceEntityDescription(
+        key="Humidity",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.HUMIDITY,
+        value_fn=lambda arve_data: arve_data.humidity,
+    ),
+    ArveDeviceEntityDescription(
+        key="PM10",
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        device_class=SensorDeviceClass.PM10,
+        value_fn=lambda arve_data: arve_data.pm10,
+    ),
+    ArveDeviceEntityDescription(
+        key="PM25",
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        device_class=SensorDeviceClass.PM25,
+        value_fn=lambda arve_data: arve_data.pm25,
+    ),
+    ArveDeviceEntityDescription(
+        key="Temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        value_fn=lambda arve_data: arve_data.temperature,
+    ),
+    ArveDeviceEntityDescription(
+        key="TVOC",
+        translation_key="tvoc",
+        value_fn=lambda arve_data: arve_data.tvoc,
+    ),
```

Comment:
I think you want to add the state class measurement here to keep track of the values as statistics